examples/platform/silabs/efr32/rs911x/wfx_rsi_host.cpp

```diff
@@ -0,0 +1,425 @@
+ * @brief
+ * Start a JOIN command to the AP - Done by the wfx_rsi task
+ * @param[in]   None
+ * @return  returns SL_STATUS_OK if successful,
+ *         SL_STATUS_INVALID_CONFIGURATION otherwise
+ ****************************************************************************/
+sl_status_t wfx_connect_to_ap(void)
+{
+    if (wfx_rsi.dev_state & WFX_RSI_ST_STA_PROVISIONED)
+    {
+        SILABS_LOG("%s: connecting to access point -> SSID: %s", __func__, &wfx_rsi.sec.ssid[0]);
+        xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+    }
+    else
+    {
+        SILABS_LOG("%s: error: access point not provisioned", __func__);
+        return SL_STATUS_INVALID_CONFIGURATION;
+    }
+    return SL_STATUS_OK;
+}
+
+#if SL_ICD_ENABLED
+#if SLI_SI917
+/*********************************************************************
+ * @fn  sl_status_t wfx_power_save()
+ * @brief
+ *      Implements the power save in sleepy application
+ * @param[in]  sl_si91x_ble_state : State to set for the BLE
+               sl_si91x_wifi_state : State to set for the WiFi
+ * @return  SL_STATUS_OK if successful,
+ *          SL_STATUS_FAIL otherwise
+ ***********************************************************************/
+sl_status_t wfx_power_save(rsi_power_save_profile_mode_t sl_si91x_ble_state, sl_si91x_performance_profile_t sl_si91x_wifi_state)
+{
+    return (wfx_rsi_power_save(sl_si91x_ble_state, sl_si91x_wifi_state) ? SL_STATUS_FAIL : SL_STATUS_OK);
+}
+#else  // For RS9116
+/*********************************************************************
+ * @fn  sl_status_t wfx_power_save()
+ * @brief
+ *      Implements the power save in sleepy application
+ * @param[in]  None
+ * @return  SL_STATUS_OK if successful,
+ *          SL_STATUS_FAIL otherwise
+ ***********************************************************************/
+sl_status_t wfx_power_save()
+{
+    return (wfx_rsi_power_save() ? SL_STATUS_FAIL : SL_STATUS_OK);
+}
+#endif /* SLI_SI917 */
+#endif /* SL_ICD_ENABLED */
+
+/*********************************************************************
+ * @fn  void wfx_setup_ip6_link_local(sl_wfx_interface_t whichif)
+ * @brief
+ *      Implement the ipv6 setup
+ * @param[in]  whichif:
+ * @return  None
+ ***********************************************************************/
+void wfx_setup_ip6_link_local(sl_wfx_interface_t whichif)
+{
+    /*
+     * TODO: Implement IPV6 setup, currently in wfx_rsi_task()
+     * This is hooked with MATTER code.
+     */
+}
+
+/*********************************************************************
+ * @fn  bool wfx_is_sta_connected(void)
+ * @brief
+ *      called fuction when driver is connected to STA
+ * @param[in]  None
+ * @return  returns ture if successful,
+ *          false otherwise
+ ***********************************************************************/
+bool wfx_is_sta_connected(void)
+{
+    bool status;
+    status = (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED) ? true : false;
+    return status;
```

Comment:
useless code here cane be optimized 
```suggestion
    return (wfx_rsi.dev_state & WFX_RSI_ST_STA_CONNECTED);
```

Reply:
Function name removed.